pkg/controller/disruption/disruption_test.go

```diff
@@ -390,30 +415,41 @@ func TestMultipleControllers(t *testing.T) {
 	pdb, pdbName := newPodDisruptionBudget(t, intstr.FromString("1%"))
 	add(t, dc.pdbStore, pdb)
 
+	pods := []*v1.Pod{}
 	for i := 0; i < podCount; i++ {
 		pod, _ := newPod(t, fmt.Sprintf("pod %d", i))
+		pods = append(pods, pod)
 		add(t, dc.podStore, pod)
 	}
+
 	dc.sync(pdbName)
 
 	// No controllers yet => no disruption allowed
 	ps.VerifyDisruptionAllowed(t, pdbName, 0)
 
 	rc, _ := newReplicationController(t, 1)
 	rc.Name = "rc 1"
+	for i := 0; i < podCount; i++ {
+		updatePodOwnerToRc(t, pods[i], rc)
```

Comment:
Can this test be split into multiple test cases, so we don't modify pods that were already added to the pod store? In my experience, that can lead to flaky/racy tests because you are changing things you already gave to the pod store, which it is not designed to support.

If these need to be in one test case/func, is it possible to add new pods rather than changing existing ones?